enumerable.rb

```diff
@@ -0,0 +1,108 @@
+module Enumerable
+  def my_each
+    i = 0
+    while i < self.size
+      yield(self[i])
+      i += 1
+    end
+    self
+  end
+
+  def my_each_with_index
+    (self.size - 1).times do |i|
+      yield(self[i], i)
+    end
+    self
+  end
+
+  def my_select
+    item = []
+    self.my_each do |i|
+      item << i if yield(i)
+    end
+    item
+  end
+
+  def my_all?
+    item = []
+    self.my_each do |i|
+      item << i if yield(i)
+    end
+
+    return true if item.size == self.size
+    return
+  end
+
+  def my_any?
+    item = []
+    self.my_each do |i|
+      item << i if yield(i)
+    end
+
+    return true if item.size > 0
+    return
+  end
+
+  def my_none?
+    item = []
+
+    self.my_each do |i|
+      item << i if yield(i)
+    end
+
+    return true if item.size == 0
+    return
+  end
+
+  def my_count(*args, &block)
+    count = 0
+
+    self.my_each do |i|
+      if args.empty? && !block_given?
+        count += 1
+      elsif block_given?
+        count += 1 if yield(i)
+      else
+        count += 1 if i.include? args
+      end
+    end
+
+    return count
+  end
+
+  def my_map
+    arr = []
+
+    self.my_each do |i|
+      arr << yield(i)
+    end
+
+    arr
+  end
+
+  def my_inject
```

Comment:
* [ ] Do same for my_inject, compare results of example with results from original.
```p (5..10).my_inject { |sum, n| sum + n }
p (5..10).my_inject(1) { |product, n| product * n } #=> 151200
longest = %w{ cat sheep bear }.my_inject do |memo, word|
   memo.length > word.length ? memo : word
end
p longest ```

Reply:
Thank you